CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ option(ENABLE_COVERAGE "Build with coverage instrumentalisation" OFF)
 option(ENABLE_COMPILER_WARNINGS "Build with compiler warnings as errors" OFF)
 option(ENABLE_SANITIZER "Use memory sanitizer for Debug build" OFF)
 option(ENABLE_TESTS "Enable Valhalla tests" ON)
+option(ENABLE_WERROR "Convert compiler warnings to errors" OFF)
```

Comment:
Is it possible this could be merged with `ENABLE_COMPILER_WARNINGS`  on line 28?

Reply:
Can you clarify what you mean? Put it differently: do you feel like we shouldn't have a separate flag to enable warnings as errors? 

I'll assume that's what you meant and explain the rationale: having `-Werror` enabled by default with this flag causes the compilation to exit early, which is not desirable if you're looking to identify and resolve errors in a specific part of the code base.  Relatedly, having mandatory errors is totally fine, but we should probably use `-Werror=<warning type>` to handle that for specific warnings. This project is not likely to be generally warning-free anytime soon.

Separately, I actually just noticed the wording here: 'Build with compiler warnings as errors', which would need to change to be consistent with the change in this PR.

Reply:
My suggestion here is that we have one flag to enable warnings (we currently have >4k), and another to convert those to errors.